scripts/outer/kubectl-kubestellar-prep_for_syncer

```diff
@@ -111,89 +111,67 @@ fi
-	echo "error reading kube-bind cluster namespace from ConfigMap kbmap-$spaceID" >&2
+if ! cluster_ns_name="$(KUBECONFIG=$espw_space_config kubectl "${kubectl_flags[@]}" -n kubestellar get configmap kbmap-"$imw" -ojsonpath="{.metadata.labels['kubestellar\.io\/kube-bind-id']}")"; then
+	echo "error reading kube-bind cluster namespace from ConfigMap kbmap-$imw" >&2
 	exit 1
 fi
+
 prefixed_stname="$cluster_ns_name"-"$stname"
-stUID=$(kubectl "${kubectl_flags[@]}" get synctargets.edge.kubestellar.io $prefixed_stname -o jsonpath="{.metadata.uid}")
-mbwsname="$spaceID"-mb-"$stUID"
+stUID=$(KUBECONFIG=$espw_space_config kubectl "${kubectl_flags[@]}" get synctargets.edge.kubestellar.io $prefixed_stname -o jsonpath="{.metadata.uid}")
+mbwsname="$imw"-mb-"$stUID"
 
-espw_space_config="${PWD}/temp-space-config/spaceprovider-default-${espw}"
-kubectl-kubestellar-get-config-for-space --space-name ${espw} --provider-name default --sm-core-config $SM_CONFIG --sm-context $SM_CONTEXT --space-config-file $espw_space_config
 if ! KUBECONFIG=$espw_space_config kubectl "${kubectl_flags[@]}" get APIExport edge.kubestellar.io &> /dev/null ; then
     echo "$0: it looks like ${espw@Q} is not the edge service provider workspace" >&2
     exit 2
 fi
 
-# TODO: mailbox does not support spaces yet
-if [ "$silent" == "true" ]; then
-    kubectl ws "${kubectl_flags[@]}" "$rootws" > /dev/null
-else
-    kubectl ws "${kubectl_flags[@]}" "$rootws"
-fi
-
-if ! kubectl "${kubectl_flags[@]}" get Workspace "$mbwsname" &> /dev/null; then
+if ! KUBECONFIG=$SM_CONFIG kubectl get space -n spaceprovider-default "$mbwsname" &> /dev/null; then
 	echo "maybe the mailbox controller is slow, give it a chance"
     sleep 15
 fi
 
-if ! kubectl "${kubectl_flags[@]}" get Workspace "$mbwsname" &> /dev/null; then
+if ! KUBECONFIG=$SM_CONFIG kubectl get space -n spaceprovider-default "$mbwsname" &> /dev/null; then
 	echo "maybe the mailbox controller is slow, give it another chance"
     sleep 15
 fi
 
-if ! kubectl "${kubectl_flags[@]}" get Workspace "$mbwsname" &> /dev/null; then
+if ! KUBECONFIG=$SM_CONFIG kubectl get space -n spaceprovider-default "$mbwsname" &> /dev/null; then
     echo "$0: did not find the mailbox workspace $mbwsname; is the mailbox controller running?" >&2
     exit 5
 fi
 
 echo "now that workspace _exists_, but is it _ready_?"
 sleep 5
 
-if ! kubectl ws "${kubectl_flags[@]}" "$mbwsname" &> /dev/null; then
+TESTMB_CMD="KUBECONFIG=$SM_CONFIG kubectl get space -n spaceprovider-default "$mbwsname" -o=jsonpath='{.status.Phase}'"
+
+
+if  phase=$(TESTMB_CMD); [ "$phase" != "READY" ]; then
     echo "tried once to switch to $mbwsname"
     sleep 15
-    if ! kubectl ws "${kubectl_flags[@]}" "$mbwsname" &> /dev/null; then
+    if phase=$(TESTMB_CMD); [ "$phase" != "READY" ]; then
         echo "tried twice to switch to $mbwsname"
         sleep 15
-        kubectl ws "${kubectl_flags[@]}" "$mbwsname"
```

Comment:
I do not think the final test for readiness should be deleted.  See the `set -x` earlier?